### ISSUE TITLE
Keybase Mac app: Icon does not disappear from Dock when all windows are closed

### ISSUE THREAD

**Issue by alexreg** [keybase/keybase-issues#2724]
If Keybase.app is launched, then the main window is closed, the icon remains in the OS X Dock at the bottom of the screen. This should not happen.

**Comment by AllanMarcus** [comment]
That is standard mac behavior for many apps. Close all the windows in Word and you will see the same behavior. There are a few apps the will auto quit when the last winded is closed. Maybe it would be better for Keybase to to that wince it only has one window.

**Comment by alexreg** [comment]
The point is not about auto-quitting, but about the dockbar icon hiding when no windows are open. When Keybase initially launches, and is only present in the icon bar at the top, then there is no dockbar icon. To have to restart the app to get rid of the dockbar icon is very annoying.

**Comment by michaelherrick** [comment]
Dock icon, standard behavior, means app is still running. You can quit any app from the file menu or right click on dock icon. This is not incorrect behavior but still probably preferable for a non-document utility to auto-quit when last window is closed.

**Comment by alexreg** [comment]
I'm afraid it really is *not* correct behaviour for *system bar* apps. I have numerous other apps in my system bar (top of screen), that when they have their windows closed, no longer show up in the dock bar, or never show up in the dock bar to begin with.

**Comment by AllanMarcus** [comment]
One common solution for this is to have a "Show Menu Item" check box in the application's preferences, and also to have a "Hide Dock Icon" check box in the preferences. Logic needs to be in there to ensure the combination of hiding both cannot be allowed.

**Comment by alexreg** [comment]
This sounds quite sensible, yes.On 10 Feb 2017, at 23:57, AllanMarcus <notifications@github.com> wrote:One common solution for this is to have a "Show Menu Item" check box in the application's preferences, and also to have a "Hide Dock Icon" check box in the preferences. Logic needs to be in there to ensure the combination of hiding both cannot be allowed.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/keybase/keybase-issues","title":"keybase/keybase-issues","subtitle":"GitHub repository","main_image_url":"<a href="https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png" class="">https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png</a>","avatar_image_url":"<a href="https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png" class="">https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png</a>","action":{"name":"Open in GitHub","url":"<a href="https://github.com/keybase/keybase-issues" class="">https://github.com/keybase/keybase-issues</a>"}},"updates":{"snippets":[{"icon":"PERSON","message":"@AllanMarcus in #2724: One common solution for this is to have a \"Show Menu Item\" check box in the application's preferences, and also to have a \"Hide Dock Icon\" check box in the preferences. Logic needs to be in there to ensure the combination of hiding both cannot be allowed. "}],"action":{"name":"View Issue","url":"<a href="https://github.com/keybase/keybase-issues/issues/2724#issuecomment-279098922" class="">https://github.com/keybase/keybase-issues/issues/2724#issuecomment-279098922</a>"}}}

**Comment by michaelherrick** [comment]
On closer look, I think you're right. Quitting the app kills the menubar icon, which is not desired. Probably should simply suppress the dock icon. Yes, you could offer a preference to show dock icon OR the menubar icon. My preference would be fewer prefs but the current user base probably would outvote me.

**Comment by AllanMarcus** [comment]
easiest might be to alway show the menu bar item when the application is running. Never show the dock icon. Have a "Quit" in the menu bar item.

**Comment by alexreg** [comment]
Right. Although oddly, I just tried it on the recent version, and the menu bar icon seems to stay running even when I quit the app… how odd!

 But yes, I’d be in favour of working a solution to this issue that minimised the number of preference options.

**Comment by alexreg** [comment]
Certainly, many apps already do this. It’s not a bad solution at all.

**Comment by TobiasBales** [comment]
This would be really great, any chance of it happening anytime soon?
